code/game/turfs/turf.dm

```diff
@@ -105,7 +105,8 @@
 					M.inertia_dir = 0
 			if(TURF_WET_LUBE)
 				M.slip(0, 4, null, (SLIDE|GALOSHES_DONT_HELP))
-				M.confused += 8
+				if(has_gravity(src))
```

Comment:
You should do if(M.slip(0, 4, null, (SLIDE|GALOSHES_DONT_HELP))) instead, which will do the proper checks for all that.